waspc/src/Analyzer/Parser/Syntax.hs

```diff
@@ -0,0 +1,76 @@
+module Analyzer.Parser.Syntax
+  ( Token (..),
+    TokenClass (..),
+    AST (..),
+    Stmt (..),
+    Expr (..),
+    ExtImportName (..),
+    Ident,
+    ParseError (..),
+    Posn (..),
+  )
+where
+
+-- TOKEN TYPES
+
+data TokenClass
+  = TLCurly
+  | TRCurly
+  | TComma
+  | TColon
+  | TLSquare
+  | TRSquare
+  | TImport
+  | TFrom
+  | TString String
+  | TInt Integer
+  | TDouble Double
+  | TTrue
+  | TFalse
+  | -- | Open quote tag, contents, Close quote tag
+    TQuoter (String, String, String)
+  | TIdent String
+  | TEOF
+  deriving (Eq, Show)
+
+data Token = Token
```

Comment:
That makes sense, if you look at it that way, and we can agree to look at it that way and keep it as it is.

On the other hand, it is not the natural terminology that comes to me, for me it is natural to think of abstract meaning of token to be Token (number 4, ...), while that blob of text inside the source (with extra metadata like position) is "token + metadata", or "token in source", or "scanned token" -> "token in context".

When you are describing it above, you say:
1. Token -> specific blob of text within the source (some string at some position)
2. TokenClass -> abstract meaning of a token
So if I simplify that (which might be silly hm), then it seems like you are saying Token is text in source, while TokenClass is token. So that would be going against the names themselves. But as I said maybe I oversimplified here.

To conclude, it is not clear to me what is the best naming. I like the following two best right now:
1. TokenType and Token
2. Token and ParsedToken

For practical reasons and experience from doing this in hlox, I would agree on going with the first one then, as `Token` is shorter than `ParsedToken` and most of the tools / libraries / literature do call this object just `Token` (because they are usually using OO languages where there is no composition like here so they don't need to come up with two names).

Reply:
More context on token/scanned token/token type: https://en.wikipedia.org/wiki/Type%E2%80%93token_distinction

And TokenClass has been renamed to TokenType.

Reply:
I read that one (partially) but didn't find it to help with the decision making here - did you find it helpful?
Ok regarding TokenType!

Reply:
I just wanted to double check that the TokenType/Token naming we're going with is the standard naming for those two concepts.